h/services/group_members.py

```diff
@@ -121,7 +122,12 @@ def member_join(self, group, userid) -> GroupMembership:
             # The user is already a member of the group.
             return existing_membership
 
-        membership = GroupMembership(group=group, user=user)
+        kwargs = {"group": group, "user": user}
+
+        if roles is not None:
+            kwargs["roles"] = roles
```

Comment:
I don't want to specify a default role here since that would be duplicated [the default that's already in the DB schema](https://github.com/hypothesis/h/blob/d09cf1223a933b2de7e246365a821a9ca11b0fe1/h/models/group.py#L78). Also don't want to do `GroupMembership(..., roles=None)` since that tries to create a row with `roles=NULL` and crashes due to a DB constraint.